### DISCUSSION TITLE
Public OAuth2 support for OVH API

### DISCUSSION THREAD

**Discussion by lalmeras** [ovh/manager#9066] [General]
Hello,

New console preview https://api.ovh.com/console-preview/ seems to use a oauth2 authentication workflow, in place of the OVH custom authentication described here https://docs.ovh.com/fr/api/first-steps-with-ovh-api/.

It seems that clientId/clientSecret for this oauth2 workflow are different than the ones created by https://eu.api.ovh.com/createToken/ (I get an `invalid_client` error if I use createToken credentials with oauth2 setting).

Is OAuth2 workflow available for custom applications or is it only restricted to OVH internal applications ? Is it planned to make it publicly available ?

**Comment by VincentCasse** [comment]
Hey @lalmeras,

Oauth2 credential will be available with the public beta of our new IAM that will comes soon!

Best regards
Vincent
